tools/api_boost/api_boost.py

```diff
@@ -0,0 +1,153 @@
+  # rewrite capabilities yet in the API booster, so we rewrite here in Python
+  # below.
+  inferred_api_includes = sorted(set(result.stdout.decode('utf-8').splitlines()))
+
+  # We just dump the inferred API header includes at the start of the #includes
+  # in the file and remove all the present API header includes. This does not
+  # match Envoy style; we rely on later invocations of fix_format.sh to take
+  # care of this alignment.
+  output_lines = []
+  include_lines = ['#include "%s"' % f for f in inferred_api_includes]
+  input_text = pathlib.Path(path).read_text()
+  for line in input_text.splitlines():
+    if include_lines and line.startswith('#include'):
+      output_lines.extend(include_lines)
+      include_lines = None
+    # Exclude API includes, except for a special case related to v2alpha
+    # ext_authz; this is needed to include the service descriptor in the build
+    # and is a hack that will go away when we remove v2.
+    if re.match(API_INCLUDE_REGEX, line) and 'envoy/service/auth/v2alpha' not in line:
+      continue
+    output_lines.append(line)
+
+  # Write to temporary file. We can't overwrite in place as we're executing
+  # concurrently with other ApiBoostFile() invocations that might need the file
+  # we're writing to.
+  pathlib.Path(path + TMP_SWP_SUFFIX).write_text('\n'.join(output_lines) + '\n')
+
+
+# Replace the original file with the temporary file created by ApiBoostFile()
+# for a given path.
+def SwapTmpFile(path):
+  pathlib.Path(path + TMP_SWP_SUFFIX).rename(path)
+
+
+# Update the Envoy source tree the latest API.
+def ApiBoostTree(args):
+  # Optional setup of state. We need the API type database, compilation database
+  # and api_booster tool in place before we can start boosting.
+  if args.generate_api_type_database:
+    sp.run(['./tools/gen_api_type_database.sh'], check=True)
+
+  if args.generate_compilation_database:
+    sp.run(['./tools/gen_compilation_database.py', '--run_bazel_build', '--include_headers'],
+           check=True)
+
+  if args.build_api_booster:
+    # Similar to gen_compilation_database.py, we only need the cc_library for
+    # setup. The long term fix for this is in
+    # https://github.com/bazelbuild/bazel/issues/9578.
+    dep_build_targets = [
+        '//source/...',
+        '//test/...',
+    ]
+    query = 'kind(cc_library, {})'.format(' union '.join(dep_build_targets))
+    dep_lib_build_targets = sp.check_output(['bazel', 'query', query]).decode().splitlines()
+    sp.run([
+        'bazel',
+        'build',
+        '--strip=always',
+        '//tools/clang_tools/api_booster',
+    ] + dep_lib_build_targets,
+           check=True)
+
+  # Figure out where the LLVM include path is. We need to provide this
+  # explicitly as the api_booster is built inside the Bazel cache and doesn't
+  # know about this path.
+  # TODO(htuch): this is fragile and depends on Clang version, should figure out
```

Comment:
Do you know what library you need from `clang/9.0.0/include`? The versioned include only have headers for compiler-rt, so I doubt you actually need this.

Reply:
If I don't declare this as `-isystem`, I see:
```
...clang-9.0/lib/clang/9.0.0/include/mmintrin.h:50:12: error: use of old-style cast [-Werror,-Wold-style-cast]
    return (__m64)__builtin_ia32_vec_init_v2si(__i, 0);    
```
since it doesn't recognize these libraries as system libs (and hence subject to relaxation of the warning).